### PR TITLE
Add maintenance windows

### LINKED ISSUE #338: [go-scalingo] Maintenance windows should be included in the database model

### PR DESCRIPTION
**Pull request by SCedricThomas**
fix #338

- [x] Add a [changelog entry](https://changelog.scalingo.com/)

### CODE REVIEW

File: databases.go [hunk 1]
```diff
@@ -229,3 +231,22 @@ func (c *Client) DatabaseDisableFeature(ctx context.Context, app, addonID, featu
...
+		"database": map[string]interface{}{
+			"maintenance_window": params,
+		},
+	}, &dbRes)
+
+	if err != nil {
+		return Database{}, errgo.Notef(err, "fail to update maintenance window")
+	}
+	return dbRes.Database, nil
+}
```
Comment (EtienneM) [inline databases.go:252]: code organization question: I would move this into the `maintenance.go` file, what do you think? It seems more in line with other features, such as backup

File: databases.go [hunk 2]
```diff
@@ -229,3 +231,22 @@ func (c *Client) DatabaseDisableFeature(ctx context.Context, app, addonID, featu
...
+func (c *Client) DatabaseUpdateMaintenanceWindow(ctx context.Context, app, addonID string, params MaintenanceWindowParams) (Database, error) {
+	var dbRes DatabaseRes
+	err := c.DBAPI(app, addonID).ResourceUpdate(ctx, "databases", addonID, map[string]interface{}{
+		"database": map[string]interface{}{
+			"maintenance_window": params,
+		},
+	}, &dbRes)
+
+	if err != nil {
+		return Database{}, errgo.Notef(err, "fail to update maintenance window")
```
Comment (EtienneM) [inline databases.go:249]:
```suggestion
		return Database{}, errgo.Notef(err, "update database '%v' maintenance window", addonID)
```

**Review by EtienneM — APPROVED** [review]

File: databases.go [hunk 3]
```diff
@@ -70,6 +71,7 @@ type Database struct {
 	Cluster                    bool              `json:"cluster"`
 	PeriodicBackupsEnabled     bool              `json:"periodic_backups_enabled"`
 	PeriodicBackupsScheduledAt []int             `json:"periodic_backups_scheduled_at"` // Hours of the day of the periodic backups (UTC)
+	MaintenanceWindow          MaintenanceWindow `json:"maintenance_window"`
```
Comment (josephpage) [inline databases.go:74]: @SCedricThomas The property `maintenance_window` is not mentionned in the documentation : https://developers.scalingo.com/databases/databases
Is it normal ?

Comment (SCedricThomas) [reply databases.go:74]: This feature is still in work in progress and not used on production at the moment but you are right! I will make an update on the documentation as soon as possible

**Pull request merged**